packages/itwinui-react/src/core/Menu/MenuItem.tsx

```diff
@@ -206,6 +195,11 @@ if (process.env.NODE_ENV === 'development') {
   MenuItem.displayName = 'MenuItem';
 }
 
+const MenuItemBase = polymorphic(
+  ButtonBase,
+  ListItem,
+) as PolymorphicForwardRefComponent<'button', ListItemOwnProps>;
```

Comment:
@r100-stack Do you think this `polymorphic` call makes sense?

An alternative would be to do this:

```jsx
<ListItem
  as={[ButtonBase, asProp] as any}
  // …
/>
```

In both cases, it's difficult or impossible to get proper types, but the `polymorphic` call at least allows defining custom types.

Reply:
> The two approaches look good from a usage perspective! 

I'm asking if it makes sense to keep `polymorphic` or do we want to use arrays. I'm leaning towards the latter to simplify things.

> In the above example, what will be the rendered element? `li`, `button`, or `asProp`?

The last one should win. `ListItem` will first render `ButtonBase` which will then render `asProp` if passed. If there is no `asProp`, then `ButtonBase` will render its own element (i.e. `<button>`).

> Could you elaborate about the types you mentioned?

When using the `as` prop, normally the prop types are automatically merged. So `<ListItem as={ButtonBase}>` provides autocomplete for props from both `ListItem` and `ButtonBase`.

But since there can be any number of components in the `as` chain, we cannot automatically merge the types. But since it is for internal implementation, I think that's ok. We can live without the autocomplete or use type assertions when necessary.

> props exposed will be a combination of a native button + ListItemOwnProps but without the props from `ButtonBase`?

Correct. We are explicitly setting the types, so if we want `ButtonBase` props, we'll have to add it manually.

> Does this approach help in the case where two props have the same name but different types?

It doesn't. For that we'd have to use a different approach, like the `asChild` or `render` prop approaches mentioned in the PR description.